checker/env.go

```diff
@@ -65,6 +73,9 @@ func (e *Env) Add(decls ...*expr.Decl) {
 	}
 }
 
+// addOverload adds overload to function declaration f.
```

Comment:
For future reference only the functions that start with a capital letter need a comment.

Reply:
Right - but comments aren't prohibited for private functions either.  In this case, the error behavior was non-obvious, to me at least, so I captured it in documentation.